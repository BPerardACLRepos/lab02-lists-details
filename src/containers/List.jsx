import React from 'react';
import { getCharacters } from '../services/avatarApi';
import Characters from '../components/character/Characters';

export default class List extends Component {
    state = {
        loading: true,
        characters: [],
    }

    async componentDidMount() {
        const characters = await getCharacters();
        this.setState({
            loading: false,
            characters,
        });
    }

    render() {
        const { loading, characters } = this.state;

        return (
            <>
                {loading &&
                    <h2>Loading...</h2>
                }
                {!loading &&
                    <Characters characters={characters} />
                }
            </>
        );
    }
}