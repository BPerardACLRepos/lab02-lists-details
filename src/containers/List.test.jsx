import React from 'react';
import { render, screen } from '@testing-library/react';
import List from './List';

describe('Avatar List Container', () => {
    it('Renders Avatar 20 character list to screen', async () => {
        render(<List />);

        screen.getByText('Loading...');

        const ul = await screen.findByRole('list', { name: 'characters' });
        expect(ul).toMatchSnapshot();
    });
});