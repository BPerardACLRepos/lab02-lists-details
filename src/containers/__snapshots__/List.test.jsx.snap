// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar List Container Renders Avatar 20 character list to screen 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <a
      href="/5cf5679a915ecad153ab68d1"
    >
      <figure>
        <img
          alt="Analay"
          src="https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201"
        />
        <figcaption>
          <h3>
            Analay
          </h3>
          <p>
            5cf5679a915ecad153ab68d1
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68d5"
    >
      <figure>
        <img
          alt="Arik"
          src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Arik.png/revision/latest?cb=20140808184002"
        />
        <figcaption>
          <h3>
            Arik
          </h3>
          <p>
            5cf5679a915ecad153ab68d5
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68e2"
    >
      <figure>
        <img
          alt="Baatar Jr."
          src="https://vignette.wikia.nocookie.net/avatar/images/4/45/Baatar_Jr..png/revision/latest?cb=20150409193510"
        />
        <figcaption>
          <h3>
            Baatar Jr.
          </h3>
          <p>
            5cf5679a915ecad153ab68e2
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68e9"
    >
      <figure>
        <img
          alt="Bolin"
          src="https://vignette.wikia.nocookie.net/avatar/images/d/d4/Bolin.png/revision/latest?cb=20150406232314"
        />
        <figcaption>
          <h3>
            Bolin
          </h3>
          <p>
            5cf5679a915ecad153ab68e9
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68fd"
    >
      <figure>
        <img
          alt="Chan (Fire Nation admiral)"
        />
        <figcaption>
          <h3>
            Chan (Fire Nation admiral)
          </h3>
          <p>
            5cf5679a915ecad153ab68fd
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab6900"
    >
      <figure>
        <img
          alt="Chit Sang"
          src="https://vignette.wikia.nocookie.net/avatar/images/8/81/Chit_Sang.png/revision/latest?cb=20121021000707"
        />
        <figcaption>
          <h3>
            Chit Sang
          </h3>
          <p>
            5cf5679a915ecad153ab6900
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68d7"
    >
      <figure>
        <img
          alt="Asami Sato"
          src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Asami_Sato.png/revision/latest?cb=20150407110150"
        />
        <figcaption>
          <h3>
            Asami Sato
          </h3>
          <p>
            5cf5679a915ecad153ab68d7
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68f0"
    >
      <figure>
        <img
          alt="Bum-Ju"
          src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Bum-Ju.png/revision/latest?cb=20140730103156"
        />
        <figcaption>
          <h3>
            Bum-Ju
          </h3>
          <p>
            5cf5679a915ecad153ab68f0
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68f5"
    >
      <figure>
        <img
          alt="Butler"
          src="https://vignette.wikia.nocookie.net/avatar/images/8/82/Butler.png/revision/latest?cb=20121107115044"
        />
        <figcaption>
          <h3>
            Butler
          </h3>
          <p>
            5cf5679a915ecad153ab68f5
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68ed"
    >
      <figure>
        <img
          alt="Broadsword man"
          src="https://vignette.wikia.nocookie.net/avatar/images/9/98/Broadsword_man.png/revision/latest?cb=20130630171553"
        />
        <figcaption>
          <h3>
            Broadsword man
          </h3>
          <p>
            5cf5679a915ecad153ab68ed
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68cf"
    >
      <figure>
        <img
          alt="Aiwei"
          src="https://vignette.wikia.nocookie.net/avatar/images/0/07/Aiwei_reveals_Korra%27s_lie.png/revision/latest?cb=20140808130551"
        />
        <figcaption>
          <h3>
            Aiwei
          </h3>
          <p>
            5cf5679a915ecad153ab68cf
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68d2"
    >
      <figure>
        <img
          alt="Animal control officer"
          src="https://vignette.wikia.nocookie.net/avatar/images/5/58/Animal_control_officer.png/revision/latest?cb=20140504160655"
        />
        <figcaption>
          <h3>
            Animal control officer
          </h3>
          <p>
            5cf5679a915ecad153ab68d2
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab6904"
    >
      <figure>
        <img
          alt="Chow"
          src="https://vignette.wikia.nocookie.net/avatar/images/5/51/Chow.png/revision/latest?cb=20140730173827"
        />
        <figcaption>
          <h3>
            Chow
          </h3>
          <p>
            5cf5679a915ecad153ab6904
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab6906"
    >
      <figure>
        <img
          alt="Circus master"
          src="https://vignette.wikia.nocookie.net/avatar/images/a/a5/Circus_master.png/revision/latest?cb=20130706153819"
        />
        <figcaption>
          <h3>
            Circus master
          </h3>
          <p>
            5cf5679a915ecad153ab6906
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68ee"
    >
      <figure>
        <img
          alt="Bujing"
          src="https://vignette.wikia.nocookie.net/avatar/images/7/76/Bujing.png/revision/latest?cb=20130714152817"
        />
        <figcaption>
          <h3>
            Bujing
          </h3>
          <p>
            5cf5679a915ecad153ab68ee
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68f6"
    >
      <figure>
        <img
          alt="Buttercup Raiko"
          src="https://vignette.wikia.nocookie.net/avatar/images/e/ed/Buttercup_Raiko.png/revision/latest?cb=20150409193510"
        />
        <figcaption>
          <h3>
            Buttercup Raiko
          </h3>
          <p>
            5cf5679a915ecad153ab68f6
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68f7"
    >
      <figure>
        <img
          alt="Cabbage merchant"
          src="https://vignette.wikia.nocookie.net/avatar/images/2/2f/Cabbage_merchant.png/revision/latest?cb=20140112200908"
        />
        <figcaption>
          <h3>
            Cabbage merchant
          </h3>
          <p>
            5cf5679a915ecad153ab68f7
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68f8"
    >
      <figure>
        <img
          alt="Calm man"
          src="https://vignette.wikia.nocookie.net/avatar/images/b/b6/Calm_man.png/revision/latest?cb=20130814093330"
        />
        <figcaption>
          <h3>
            Calm man
          </h3>
          <p>
            5cf5679a915ecad153ab68f8
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab68ff"
    >
      <figure>
        <img
          alt="Chin"
          src="https://vignette.wikia.nocookie.net/avatar/images/6/6f/Chin_the_Great.png/revision/latest?cb=20140215111604"
        />
        <figcaption>
          <h3>
            Chin
          </h3>
          <p>
            5cf5679a915ecad153ab68ff
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
  <li>
    <a
      href="/5cf5679a915ecad153ab6901"
    >
      <figure>
        <img
          alt="Chit Sang's friend"
          src="https://vignette.wikia.nocookie.net/avatar/images/6/6e/Chit_Sang%27s_friend.png/revision/latest?cb=20121120131829"
        />
        <figcaption>
          <h3>
            Chit Sang's friend
          </h3>
          <p>
            5cf5679a915ecad153ab6901
          </p>
        </figcaption>
      </figure>
    </a>
  </li>
</ul>
`;
